### PR TITLE
Fix #30 Add zapcore.ArrayMarshalerType encoder

### LINKED ISSUE #30: otelzap zapcore.ArrayMarshalerType is not implemented
The current implementation silently drops a field without any errors.

https://github.com/uptrace/opentelemetry-go-extra/blob/479240184b44f36f1623c6a3c5426e5ff0468c25/otelzap/otelzap.go#L688

### PR DESCRIPTION
**Pull request by ezh**
This MR fixes #30. The commit message is pretty clear. Feel free to update the wording.

I added 2 test cases: for the simplest case with an array of strings and the complex one with an array of durations. IMO other cases should work without any issues as well.

### CODE REVIEW

File: otelzap/arrayEncoder.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
+package otelzap
+
+import (
+	"fmt"
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+type bufferArrayEncoder struct {
```
Comment (vmihailenco) [inline otelzap/arrayEncoder.go:10]: Could you add a brief comment how this works? I assume it implements some interface from zap. If so, we should also add a compile-time check like `var _ zap.SomeInterface = (*bufferArrayEncoder)(nil)`

File: otelzap/arrayEncoder.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+func (t *bufferArrayEncoder) AppendInt16(v int16) {
+	t.stringsSlice = append(t.stringsSlice, fmt.Sprintf("%v", v))
+}
+
+func (t *bufferArrayEncoder) AppendInt8(v int8) {
+	t.stringsSlice = append(t.stringsSlice, fmt.Sprintf("%v", v))
+}
+
+func (t *bufferArrayEncoder) AppendString(v string) {
+	t.stringsSlice = append(t.stringsSlice, fmt.Sprintf("%v", v))
```
Comment (vmihailenco) [inline otelzap/arrayEncoder.go:82]: `t.stringsSlice = append(t.stringsSlice, v)` to make at least this case fast

**Comment by vmihailenco** [comment]
Thanks. Using `fmt.Sprintf` is not very efficient but I guess we can improve this later. I've left 2 small comments, but otherwise this looks good.

Thanks for the PR :+1:

File: otelzap/arrayEncoder.go (hunk 1 above)
Comment (ezh) [reply otelzap/arrayEncoder.go:13]: Done. Description and validation are added.

File: otelzap/arrayEncoder.go (hunk 2 above)
Comment (ezh) [reply otelzap/arrayEncoder.go]: Fixed. Thanks. I was focused on the goal.

**Comment by ezh** [comment]
@vmihailenco I guess that efficiency improvement should be delivered as several benchmark tests to compare exact numbers. It is a separate task for someone who has spare time for that 😉 

BTW I checked repository PRs and found that they all contain only one commit. What is the proper commit message for the second commit? @vmihailenco could you please help with that. I could squash history as well.

**Comment by vmihailenco** [comment]
You can use `chore: ` prefix to omit the commit for the changelog. But I can just use `Squash and merge` button instead.

Thanks for the PR :+1:

**Pull request merged**